File: Day6-FindAllDuplicatesInAnArray/Solution.cpp
// Runtime: 112 ms
// Memory Usage: 33.8 MB

class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> result;
        for(int i=0;i<nums.size();i++){
            if(nums[abs(nums[i])-1] < 0){
                result.push_back(abs(nums[i]));
            }
            else{
                nums[abs(nums[i])-1] *= -1;
            }
        }
        return result;
    }
};

File: Day6-FindAllDuplicatesInAnArray/Solution.py
# Runtime: 332 ms
# Memory Usage: 22.8 MB

class Solution(object):
    def findDuplicates(self, nums):
        dict1 = {}
        for i in nums:
            if i in dict1:
                dict1[i]+=1
            else:
                dict1[i]=1
        return [i for i in dict1 if dict1[i]==2]
